### ISSUE TITLE
9.2.4.4: Gelten Sprachlinks wie z.B. 'DE' und 'EN' als aussagekräftig?

### ISSUE THREAD

**Issue by cstrobbe** [BIK-BITV/BIK-Web-Test#64]
Auf vielen Websites kann man die Sprache wechseln mithilfe von Links die mit "DE", "EN" oder ähnlichen Abkürzungen beschriftet sind. Bis jetzt habe ich solche Linktext als nicht aussagekräftig betrachtet.
(Ob bei diesen Links auch noch ein `title`-Attribut eingesetzt wird oder nicht, macht aus meiner Sicht keinen Unterschied. Siehe [Using the HTML title attribute – Updated March 2020](https://developer.paciellogroup.com/blog/2010/11/using-the-html-title-attribute/) von Steve Faulkner.)
Nicht alle ISO-639-Tags (wenn die Abkürzungen tatsächlich auf diesen Tags basieren) sind leicht zu verstehen, z.B. `ga` für Irisch oder `el` für Griechisch (um nur offiziele Sprachen in der EU zu nennen ...).

Ist diese Bewertung konsistent mit den anderen Prüfstellen?

**Comment by stefanFarnetani** [comment]
Man muss einräumen, dass derjenige Website-Nutzer der griechisch benötigt, meist auch weiß wie griechisch international abgekürzt wird. Für einen Griechen wäre das deutsche Wort `griechisch` eher eine Hürde und schwieriger zu verstehen als `el`.
Deshalb bewerte ich die ISO-Tags als Sprachnavigation auf einer Website als aussagekräftig gewertet.

Anders verhält es sich in einer Applikation in der man z.B. zusätzliche Sprachbibliotheken installieren kann. Dort erwarte ich die ausgeschriebene Sprache in der jeweiligen Sprache des Benutzers.

**Comment by johannesFischer84** [comment]
> Auf vielen Websites kann man die Sprache wechseln mithilfe von Links die mit "DE", "EN" oder ähnlichen Abkürzungen beschriftet sind. Bis jetzt habe ich solche Linktext als nicht aussagekräftig betrachtet.

Die WCAG lassen in 2.4.4 eine Ausnahme zu. Wenn ein Linktext für Nutzer allgemein unverständlich ist, ist das Kriterium dennoch konform, siehe folgendes Zitat:
> except where the purpose of the link would be ambiguous to users in general

Aus meiner Sicht kann man bei Sprachabkürzungen nur den Hinweis geben, dass es sinnvoller wäre, sie auszuschreiben. Fehlerhaft wären die Bezeichnungen wahrscheinlich eher nach Kriterium 3.1.4 (Abkürzungen), das aber auf Level AAA steht.

Allerdings würde ich es begrüßen, wenn die WCAG-Ausnahme der Unverständlichkeit für alle Nutzer in der Prüfanleitung ergänzt werden könnte.

**Comment by detlevhfischer** [comment]
Ich stimme Stefan zu. Im Kontext von Sprach 
-Auswahlmenüs würde ich auch an FR, DE, EN nicht als FAIL von 2.4.4 betrachten, auch wenn ein title oder abbr-Element empfehlenswert wäre.

**Comment by johannesFischer84** [comment]
@detlevhfischer Detlev, ich denke, es kommt darauf an. Schau mal auf folgende Seite: https://www.dhmd.de
Da ist im Kopfbereich eine Sprachauswahl, wobei eine spitze Klammer zu sehen ist und dann ein Dropdown. In diesem Fall passt das nicht auf Prüfschritt 2.4.4, weil hier im HTML ein `select `und keine Links verwendet wurden, aber nimm mal an, es sind Links, das versteckte `label` mit der Sprachauswahl ist nicht da, es sieht aber optisch gleich aus. Ich finde, die Linktexte de, en, cs usw. wären für sehende und nicht sehende Menschen gleich schlecht aussagekräftig. Ich würde das nach der Ausnahme der WCAG in 2.4.4 trotzdem als konform bewerten, weil ich es für alle Nutzer unverständlich finde.
Zumindest für unerfahrene Nutzer, das ist wieder so ein wenig so eine Sache wie mit dem Stern-Zeichen als Pflichtfeldkennzeichner von Eingabefeldern, ob und wo man ihn erklären muss.
Wenn du es anders siehst, höre ich gern deine Meinung.

**Comment by cstrobbe** [comment]
Braucht man auf Basis dieser Diskussion keinen Hinweis in der Prüfschrittbeschreibung, der darauf hinweist, das Links die für alle Benutzer gleich wenig aussagekräftig sind, keinen Verstoß darstellen?

**Comment by johannesFischer84** [comment]
@cstrobbe Stimmt, bei 2.4.6 ist keine solche Ausnahme vorhanden. Mich hat diese Inkonsistenz von 2.4.4 und 2.4.6 auch schon immer gewundert. Vielleicht wurde bei Links eher beobachtet, dass sie an sich zumindest allein (bei 2.4.9 auf AAA-Level gibt es die Ausnahme weiterhin) häufiger wenig aussagekräftig sind als Überschriften und Beschriftungen von Eingabefeldern/Schaltern. Aber das ist spekulativ.
